addmeta/addmeta.py

```diff
@@ -169,4 +173,23 @@ def list_from_file(fname):
     with open(fname, 'rt') as f:
         filelist = [Path(fname).parent / file for file in skip_comments(f)]
 
-    return filelist
\ No newline at end of file
+    return filelist
+
+def sort_metadict_remove_attrs(metadict, rootgrp):
+    """
+    Add all global attributes from rootgrp to the metadict, sort them, remove
+    the attributes from rootgrp and then return the new metadict
+    """
+    # Add all existing attrs to the metadict (if they're not already there)
+    for attr in rootgrp.ncattrs():
+        if attr not in metadict['global']:
+            metadict['global'][attr] = rootgrp.getncattr(attr)
+
+    # Sort metadict
+    metadict['global'] = {attr: metadict['global'][attr] for attr in sorted(metadict['global'].keys(), key=str.casefold)}
+
+    # Remove all existing attrs from the rootgrp
+    for attr in rootgrp.ncattrs():
+        rootgrp.delncattr(attr)
+
+    return metadict
```

Comment:
```suggestion
def delete_global_metdata(metadict):
    """
    Delete all global attributes and return as dict 
    """
    deleted = {}

    # Remove all existing attrs from the rootgrp
    for attr in rootgrp.ncattrs():
        deleted[attr] = rootgrp.getncattr(attr)
        rootgrp.delncattr(attr)

    return deleted
```

I'd argue modifying `rootgrp` is close to a [side-effect](https://medium.com/@ryk.kiel/dont-let-your-code-get-out-of-control-avoiding-side-effects-in-python-d68faf26912).

I think it would be cleaner to separate this into a function that just deletes all the global attributes and returns a `dict` (see above), then `update` *that* dict with `metadict` (which will overwrite any shared keys) and sort (see above).

Reply:
Good suggestion! I like it.